### PR TITLE
Accessing shared dashboards via API

### PR DESCRIPTION
**Pull request by rajsarkapally-sfdc**

### CODE REVIEW

File: ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java [hunk 1]
```diff
@@ -67,202 +67,225 @@
...
-                result.addAll(dService.findDashboards(null));
-            } else {
-                result.addAll(dService.findDashboardsByOwner(owner));
-                result.addAll(dService.findSharedDashboards());
-            }
-        }
-        return result;
-    }
+		PrincipalUser owner=null, loggedInUser=getRemoteUser(req);
+		if(ownerName !=null && ownerName.length()>0){
```
Comment (dilipdevaraj-sfdc) [inline ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java]: Do we have a corresponding work item - I am trying to understand the use case a bit better - whether it is solving  a user story or bug for shared dashboards

File: ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java [hunk 2]
```diff
@@ -67,202 +67,225 @@
...
-     * @param   owner          The principlaUser owner for owner name filter.
-     *
-     * @return  The list of filtered dashboards in dashboard object.
-     */
-    private Set<Dashboard> getDashboardsObj(String dashboardName, PrincipalUser owner) {
-        Set<Dashboard> result = new HashSet<>();
-        if (dashboardName != null && !dashboardName.isEmpty()) {
-            Dashboard dashboard = dService.findDashboardByNameAndOwner(dashboardName, owner);
+	/**
+	 * Return all dashboards in dashboard objects filtered by owner.
```
Comment (rajsarkapally-sfdc) [inline ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java]: Done

File: ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java (hunk 1 above)
Comment (rajsarkapally-sfdc) [inline ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java]: https://gus.my.salesforce.com/apex/adm_bugdetail?id=a07B0000002U8R5IAK&sfdc.override=1

File: ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java [hunk 3]
```diff
@@ -67,202 +67,225 @@
...
-        return DashboardDto.transformToDtoNoContent(new LinkedList<>(result));
-    }
+		Set<Dashboard> result = new HashSet<>();
+		if (dashboardName != null && !dashboardName.isEmpty()) {
+			Dashboard dashboard = dService.findDashboardByNameAndOwner(dashboardName, owner);
+			if (dashboard != null) {
+				result.add(dashboard);
+			}
+			if(!owner.equals(loggedInUser) && !loggedInUser.isPrivileged()){
+				_removeNonSharedDashboards(result);
```
Comment (rajsarkapally-sfdc) [inline ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java]: Let user1 created a non-shared alert alert1 and a shared alert alert2, Now user2 wants to read alerts created by user1. We retrieve all user1 alerts and remove non shared alerts before returning alerts to user2

File: ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java (hunk 2 above)
Comment (dilipdevaraj-sfdc) [inline ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/DashboardResources.java]: @rajsarkapally-sfdc 

Can you edit the Java doc for this method, with "Return all dashboards in dashboard objects filtered by owner, as well as shared shared dashboards". I can then go ahead and approve. Thanks.

**Pull request merged**